daemon/mgr/network.go

```diff
@@ -332,29 +332,61 @@ func (nm *NetworkManager) EndpointCreate(ctx context.Context, endpoint *types.En
 
 // EndpointRemove is used to remove network endpoint.
 func (nm *NetworkManager) EndpointRemove(ctx context.Context, endpoint *types.Endpoint) error {
+	var (
+		ep libnetwork.Endpoint
+	)
+
 	sid := endpoint.NetworkConfig.SandboxID
 	epConfig := endpoint.EndpointConfig
 
-	logrus.Debugf("remove endpoint: %s on network: %s", epConfig.EndpointID, endpoint.Name)
+	logrus.Debugf("remove endpoint(%s) on network(%s)", epConfig.EndpointID, endpoint.Name)
 
 	if sid == "" {
 		return nil
 	}
 
+	// find endpoint in network and delete it.
 	sb, err := nm.controller.SandboxByID(sid)
-	if err == nil {
-		if err := sb.Delete(); err != nil {
-			logrus.Errorf("failed to delete sandbox id: %s, err: %v", sid, err)
-			return err
+	if err != nil || sb == nil {
+		return errors.Errorf("failed to get sandbox by id(%s)", sid)
+	}
+
+	eplist := sb.Endpoints()
+	if len(eplist) == 0 {
+		return errors.Errorf("no endpoint in sandbox(%s)", sid)
+	}
+
+	for _, e := range eplist {
+		if e.ID() == epConfig.EndpointID {
+			ep = e
+			break
 		}
-	} else if _, ok := err.(networktypes.NotFoundError); !ok {
-		logrus.Errorf("failed to get sandbox id: %s, err: %v", sid, err)
-		return fmt.Errorf("failed to get sandbox id: %s, err: %v", sid, err)
+	}
+
+	if ep == nil {
+		return errors.Errorf("not connected to the network(%s)", endpoint.Name)
+	}
+
+	if err := ep.Leave(sb); err != nil {
```

Comment:
In this pr, you said `remove all endpoints when network disconnect`, 

However, I found that it only deletes one endpoint. Is it correct?